mne/annotations.py

```diff
@@ -758,6 +762,141 @@ def rename(self, mapping, verbose=None):
         return self
 
 
+class HEDAnnotations(Annotations):
+    """Annotations object for annotating segments of raw data with HED tags.
+
+    Parameters
+    ----------
+    onset : array of float, shape (n_annotations,)
+        The starting time of annotations in seconds after ``orig_time``.
+    duration : array of float, shape (n_annotations,) | float
+        Durations of the annotations in seconds. If a float, all the
+        annotations are given the same duration.
+    description : array of str, shape (n_annotations,) | str
+        Array of strings containing description for each annotation. If a
+        string, all the annotations are given the same description. To reject
+        epochs, use description starting with keyword 'bad'. See example above.
+    hed_tags : array of str, shape (n_annotations,) | str
+        Array of strings containing a HED tag for each annotation. If a single string
+        is provided, all annotations are given the same HED tag.
+    hed_version : str
+        The HED schema version against which to validate the HED tags.
+    orig_time : float | str | datetime | tuple of int | None
+        A POSIX Timestamp, datetime or a tuple containing the timestamp as the
+        first element and microseconds as the second element. Determines the
+        starting time of annotation acquisition. If None (default),
+        starting time is determined from beginning of raw data acquisition.
+        In general, ``raw.info['meas_date']`` (or None) can be used for syncing
+        the annotations with raw data if their acquisition is started at the
+        same time. If it is a string, it should conform to the ISO8601 format.
+        More precisely to this '%%Y-%%m-%%d %%H:%%M:%%S.%%f' particular case of
+        the ISO8601 format where the delimiter between date and time is ' '.
+    %(ch_names_annot)s
+
+    See Also
+    --------
+    mne.Annotations
+
+    Notes
+    -----
+
+    .. versionadded:: 1.10
+    """
+
+    def __init__(
+        self,
+        onset,
+        duration,
+        description,
+        hed_tags,
+        hed_version="latest",  # TODO @VisLab what is a sensible default here?
+        orig_time=None,
+        ch_names=None,
+    ):
+        hed = _soft_import("hedtools", "validation of HED tags in annotations")  # noqa
+        # TODO is some sort of initialization of the HED cache directory necessary?
+        super().__init__(
+            self,
+            onset=onset,
+            duration=duration,
+            description=description,
+            orig_time=orig_time,
+            ch_names=ch_names,
+        )
+        # TODO validate the HED version the user claims to be using.
```

Comment:
@VisLab

Reply:
When the user tries to load the schema version --- if it can't find it, the load fails. This happens at the beginning of validation. For now I think that is sufficient.

Reply:
so in other words, when the `hed_strings` are passed to the validator, the schema version is also passed in (and validated). That makes sense. So no need to do a separate validation of the schema version here.